settings.gradle.kts

```diff
@@ -22,6 +22,6 @@ plugins {
     id("org.gradle.toolchains.foojay-resolver-convention") version ("0.3.0")
 }
 
-if (JavaVersion.current() < JavaVersion.VERSION_11) {
-    throw GradleException("SpongeAPI requires at least Java 11 to build, but you have ${JavaVersion.current()}.")
+if (JavaVersion.current() < JavaVersion.VERSION_17) {
```

Comment:
The Java version required to run gradle does not necessarily have to be the Java version used to compile -- this is just for buildscript plugins (iirc originally checkstyle) to run.

Reply:
Well, you can build either way -- if you're running on an older Java version, Gradle will try to detect the Java 17 JVM or if necessary download it -- it's easier to be more lenient rather than making devs muck around with their environment variables.

Reply:
I believe I added that check because newer versions of Checkstyle require J11 and Gradle didn't yet support toolchains on the checkstyle task at the time (it does now, so that could be configured to use the project toolchain instead and this check could be removed)